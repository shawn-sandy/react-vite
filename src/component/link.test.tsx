import React from 'react';

import { describe, expect, test, it  } from 'vitest';
import { render, screen  } from '@testing-library/react';
import Link from"./Link"

describe("Test link", () => {
  test("component renders as expected to", () => {
    const container = render(<Link />)
    expect(screen.getByText('Default Link')).toBeDefined()
    screen.debug()
    expect(container).toMatchSnapshot()
  })

  it("should accept props", () => {

    const link = "https://google.com"
    const label = " Search on Google"

    const linkContainer = render(<Link page={link}>{label}</Link> )

    expect(linkContainer).toMatchSnapshot()
    screen.debug()

  })
})
