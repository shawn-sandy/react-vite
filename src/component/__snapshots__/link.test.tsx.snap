// Vitest Snapshot v1

exports[`<Link> test > Link component renders as expected to 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="normal"
        href="#"
      >
        Default Link
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="normal"
      href="#"
    >
      Default Link
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Link> test > Link should accept and display props 1`] = `
<a
  class="normal"
  href="https://google.com/"
>
  Search on Google
</a>
`;
